core/controllers/editor_test.py

```diff
@@ -1832,3 +1832,134 @@ def test_discard_draft(self):
         self.assertIsNone(exp_user_data.draft_change_list)
         self.assertIsNone(exp_user_data.draft_change_list_last_updated)
         self.assertIsNone(exp_user_data.draft_change_list_exp_version)
+
+
+class HasSeenTutorialTests(BaseEditorControllerTests):
+
+    def test_get_user_has_seen_editor_tutorial(self):
+        self.login(self.OWNER_EMAIL)
+
+        exp_id = exp_services.get_new_exploration_id()
+        self.save_new_valid_exploration(exp_id, self.owner_id)
+
+        response = self.get_json(
+            '%s/%s' % (feconf.EXPLORATION_DATA_PREFIX, exp_id))
+        self.assertTrue(response['show_state_editor_tutorial_on_load'])
+
+        user_services.record_user_started_state_editor_tutorial(
+            self.owner_id)
+
+        response = self.get_json(
+            '%s/%s' % (feconf.EXPLORATION_DATA_PREFIX, exp_id))
+        self.assertFalse(response['show_state_editor_tutorial_on_load'])
+
+        self.logout()
+
+    def test_get_user_has_seen_translation_tutorial(self):
+        self.login(self.OWNER_EMAIL)
+
+        exp_id = exp_services.get_new_exploration_id()
+        self.save_new_valid_exploration(exp_id, self.owner_id)
+
+        response = self.get_json(
+            '%s/%s' % (feconf.EXPLORATION_DATA_PREFIX, exp_id))
+        self.assertTrue(response['show_state_translation_tutorial_on_load'])
+
+        user_services.record_user_started_state_translation_tutorial(
+            self.owner_id)
+
+        response = self.get_json(
+            '%s/%s' % (feconf.EXPLORATION_DATA_PREFIX, exp_id))
+        self.assertFalse(response['show_state_translation_tutorial_on_load'])
+
+        self.logout()
+
+    def test_put_user_has_seen_editor_tutorial(self):
+        self.login(self.OWNER_EMAIL)
+
+        exp_id = exp_services.get_new_exploration_id()
+        self.save_new_valid_exploration(exp_id, self.owner_id)
+
+        response = self.get_html_response('/create/%s' % exp_id)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        add_url = '%s/%s' % (feconf.EXPLORATION_DATA_PREFIX, exp_id)
+        exploration = exp_services.get_exploration_by_id(exp_id)
+        change_dict = {
```

Comment:
I'm confused by this -- maybe add a comment to explain, here and below. Basically ... what does changing the title have to do with seeing the editor tutorial?

Reply:
So you mean we should only render show_state_editor_tutorial_on_load and show_state_translation_tutorial_on_load in the GET method?

Reply:
Have you read the code? Try to figure out where these are used in the frontend, then you can make a decision based on that.

Reply:
Ok @seanlip can you guide me a bit? I am a little confused on how to figure out whether they are needed in the PUT method.

Reply:
Can you start by explaining to me where the values for these variables are used in the frontend? Include links to lines of code in files, and explain your reasoning.

Reply:
Can you start by explaining where the values for these variables are used in the frontend? Include links to lines of code in files, and explain your reasoning.

Reply:
show_state_editor_tutorial_on_load is used [here](https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/pages/exploration_editor/ExplorationEditor.ts#L385) and show_state_translation_tutorial_on_load is used [here](https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/pages/exploration_editor/ExplorationEditor.ts#L388) in frontend

Reply:
OK. Are those the only places where they are used?

If so, then these assignments only happen in initExplorationPage(). When is this function called? Does it get called after any PUT requests? Does it make any PUT requests?

Conversely, when is the PUT handler in the backend called? In those calls, what does the frontend do with the result of that call? Does it make use of the show_state_editor_tutorial_on_load etc. properties?

Reply:
I don't see any use of those two variables in the front-end PUT method. You can see [here](https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/domain/exploration/EditableExplorationBackendApiService.ts#L151). Should I remove them from the backend PUT then?

Reply:
Yes, if they are not used at all by the frontend, you can remove them from the PUT. You just need to make sure that they aren't used before doing that.

Reply:
Done!